components/accelerator/nvidia/query/pci/pci.go

```diff
@@ -0,0 +1,292 @@
+package pci
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/leptonai/gpud/pkg/file"
+	"github.com/leptonai/gpud/pkg/process"
+
+	"sigs.k8s.io/yaml"
+)
+
+// Lists all PCI devices.
+func List(ctx context.Context, opts ...OpOption) (Devices, error) {
+	op := &Op{}
+	if err := op.applyOpts(opts); err != nil {
+		return nil, err
+	}
+
+	lspciPath, err := file.LocateExecutable("lspci")
+	if err != nil {
+		return nil, nil
+	}
+	if lspciPath == "" {
+		return nil, nil
+	}
+
+	p, err := process.New(
+		process.WithBashScriptContentsToRun(fmt.Sprintf("sudo %s -vvv", lspciPath)),
+		process.WithRunAsBashScript(),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := p.Start(ctx); err != nil {
+		return nil, err
+	}
+
+	scanner := bufio.NewScanner(p.StdoutReader())
+	devs, err := parseLspciVVV(ctx, scanner, op.nameMatchFunc)
+	if err != nil {
+		return nil, err
+	}
+
+	select {
+	case err := <-p.Wait():
+		if err != nil {
+			return nil, err
+		}
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	}
+
+	return devs, nil
+}
+
+type Devices []Device
+
+func (devs Devices) JSON() ([]byte, error) {
+	return json.Marshal(devs)
+}
+
+func (devs Devices) YAML() ([]byte, error) {
+	return yaml.Marshal(devs)
+}
+
+type Device struct {
+	// ID of the PCI device.
+	// e.g., "00:0e.0" in "00:0e.0 PCI ..."
+	ID string `json:"id"`
+
+	// Name that comes after the ID.
+	// e.g., "3D controller" in "00:0e.0 3D controller"
+	Name string `json:"name"`
+
+	// Access control service.
+	// e.g., Capabilities: [170 v1] Access Control Services
+	AccessControlService *AccessControlService `json:"access_control_service,omitempty"`
```

Comment:
we want this to be disabled for BM?

Reply:
Yes

> Baremetal systems
> If PCI switches have ACS enabled, it needs to be disabled.
>
> Virtual machines
> Virtual machines require ACS to function, hence disabling ACS is not an option.

https://docs.nvidia.com/deeplearning/nccl/user-guide/docs/troubleshooting.html#pci-access-control-services-acs